plugins/nf-k8s/src/main/nextflow/k8s/K8sConfig.groovy

```diff
@@ -140,74 +286,36 @@ class K8sConfig implements Map<String,Object> {
      *
      */
     boolean entrypointOverride() {
-        def result = target.entrypointOverride
-        if( result == null )
-            result = System.getenv('NXF_CONTAINER_ENTRYPOINT_OVERRIDE')
-        return result
-    }
-
-    /**
-     * @return the path where the workflow is launched and the user data is stored
-     */
-    String getLaunchDir() {
-        if( target.userDir ) {
-            log.warn "K8s `userDir` has been deprecated -- Use `launchDir` instead"
-            return target.userDir
-        }
-        target.launchDir ?: "${getStorageMountPath()}/${getUserName()}" as String
-    }
-
-    /**
-     * @return Defines the path where the workflow temporary data is stored. This must be a path in a shared K8s persistent volume (default:<user-dir>/work).
-     */
-    String getWorkDir() {
-        target.workDir ?: "${getLaunchDir()}/work" as String
-    }
-
-    /**
-     * @return Defines the path where Nextflow projects are downloaded. This must be a path in a shared K8s persistent volume (default: <volume-claim-mount-path>/projects).
-     */
-    String getProjectDir() {
-        target.projectDir ?: "${getStorageMountPath()}/projects" as String
+        return ContainerHelper.entrypointOverride()
     }
 
-    String getNamespace() { target.namespace }
-
-    boolean useJobResource() { ResourceType.Job.name() == target.computeResourceType?.toString() }
-
-    String getServiceAccount() { target.serviceAccount }
+    boolean useJobResource() { ResourceType.Job.name() == computeResourceType }
 
     String getNextflowImageName() {
-        final defImage = "nextflow/nextflow:${BuildInfo.version}"
-        return target.navigate('nextflow.image', defImage)
-    }
-
-    boolean getAutoMountHostPaths() {
-        Boolean.valueOf( target.autoMountHostPaths as String )
+        return "nextflow/nextflow:${BuildInfo.version}"
     }
```

Comment:
There is a hidden option `k8s.nextflow.image` which controls the nextflow container image used by `kuberun`. This can already be controlled via CLI option so I don't think we need a config option for it.